### PR TITLE
Gorger's Psychic Link QoL change

### PR DESCRIPTION
**Pull request by lKiseki**
## About The Pull Request
Before, the Gorger's Psychic Link would cancel itself if the target xeno left the link's range.

With these changes, the Psychic Link will only toggle itself off if the target isn't within range, and automatically toggle on if the target reenters range.

## Why It's Good For The Game
Quality of Life changes that should make Psychic Link much more easier to use.

also slotbot told me to say that gorger is a great xeno
gorger is a great xeno

## Changelog
:cl: Lewdcifer
qol: Psychic Link will not cancel itself due to distance, instead toggles on/off
/:cl:

### CODE REVIEW

File: code/datums/status_effects/xeno_buffs.dm [hunk 1]
```diff
@@ -142,10 +141,25 @@
 ///Handles the link breaking due to distance
 /datum/status_effect/xeno_psychic_link/proc/handle_dist(datum/source)
 	SIGNAL_HANDLER
-	if(get_dist(owner, target_mob) > link_range)
-		to_chat(target_mob, span_xenowarning("You are too far away from [owner]."))
-		to_chat(owner, span_xenowarning("[target_mob] is too far away."))
-		owner.remove_status_effect(STATUS_EFFECT_XENO_PSYCHIC_LINK)
+	var/within_range_current = get_dist(owner, target_mob) <= link_range
+	if(within_range_current == within_range)
```
Comment (BraveMole) [inline code/datums/status_effects/xeno_buffs.dm:145]: i don't understand what you are trying to do

within_range_current, as you defined it, is a boolean (true of false)

You never check two booleans with ==

File: code/datums/status_effects/xeno_buffs.dm [hunk 2]
```diff
@@ -142,10 +141,25 @@
...
+		return
+	within_range = within_range_current
+	link_toggle(within_range)
+	to_chat(owner, within_range ? span_xenowarning("[target_mob] is within range again.") : span_xenowarning("[target_mob] is too far away."))
```
Comment (BraveMole) [inline code/datums/status_effects/xeno_buffs.dm:149]: pretty sure that can become very spammy for no real information given

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/xeno_buffs.dm (hunk 2 above)
Comment (lKiseki) [reply code/datums/status_effects/xeno_buffs.dm:149]: Me and Slotbot opted for keeping it this way for the sake of clarification, for both players. If it's a real concern though, we can see about making it less spammy.

File: code/datums/status_effects/xeno_buffs.dm [hunk 3]
```diff
@@ -142,10 +141,25 @@
...
 /datum/status_effect/xeno_psychic_link/proc/handle_dist(datum/source)
 	SIGNAL_HANDLER
-	if(get_dist(owner, target_mob) > link_range)
-		to_chat(target_mob, span_xenowarning("You are too far away from [owner]."))
-		to_chat(owner, span_xenowarning("[target_mob] is too far away."))
-		owner.remove_status_effect(STATUS_EFFECT_XENO_PSYCHIC_LINK)
+	var/within_range_current = get_dist(owner, target_mob) <= link_range
+	if(within_range_current == within_range)
+		return
+	within_range = within_range_current
```
Comment (BraveMole) [inline code/datums/status_effects/xeno_buffs.dm:147]:
```suggestion
	was_within_range = within_range
```
And propagate those changes

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/xeno_buffs.dm [hunk 4]
```diff
@@ -142,10 +141,25 @@
...
-		owner.remove_status_effect(STATUS_EFFECT_XENO_PSYCHIC_LINK)
+	var/within_range = get_dist(owner, target_mob) <= link_range
+	if(within_range == was_within_range)
+		return
+	was_within_range = within_range
+	link_toggle(was_within_range)
+	to_chat(owner, was_within_range ? span_xenowarning("[target_mob] is within range again.") : span_xenowarning("[target_mob] is too far away."))
+
+///Handles the link toggling on and off
+/datum/status_effect/xeno_psychic_link/proc/link_toggle(toggle)
```
Comment (BraveMole) [inline code/datums/status_effects/xeno_buffs.dm:152]:
```suggestion
/datum/status_effect/xeno_psychic_link/proc/link_toggle(toggle_on)
```

File: code/datums/status_effects/xeno_buffs.dm [hunk 5]
```diff
@@ -142,10 +141,25 @@
...
+	if(toggle)
```
Comment (BraveMole) [inline code/datums/status_effects/xeno_buffs.dm:153]:
```suggestion
	if(toggle_on)
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**